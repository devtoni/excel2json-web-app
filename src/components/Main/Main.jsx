import React, { Component } from "react";
import { Form } from "../Form/Form";
import { SaveAs } from "../SaveAs/SaveAs";
import { NoResults } from "../NoResults/NoResults";
import ReactJson from "react-json-view";
import "./Main.scss";

export class Main extends Component {
  state = { jsonFile: [] };

  _updateState = data => {
    this.setState({ jsonFile: data });
  };

  _renderJson() {
    return this.state.jsonFile.length === 0 ? (
      <NoResults />
    ) : (
      <ReactJson src={this.state.jsonFile} name={false} theme={"monokai"} />
    );
  }

  render() {
    return (
      <main className="Main">
        <div className="Main-content">
          <div className="Main-contentForm">
            <Form getJsonFile={this._updateState} />
          </div>
          <div className="Main-contentPlace">{this._renderJson()}</div>
        </div>
        <div className="Main-saveButton">
          <SaveAs data={this.state.jsonFile} />
        </div>
      </main>
    );
  }
}
