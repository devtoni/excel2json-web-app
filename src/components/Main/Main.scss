.Main {
  &-content {
    display: flex;
    flex-direction: column;
    &Form {
      display: flex;
      flex: 1;
      flex-direction: column;
      & span > div {
        min-height: 150px;
        display: flex;
        align-items: center;
        &:last-child {
          min-height: auto;
        }
      }
    }
    &Place {
      flex: 1;
      max-height: 400px;
      overflow-y: scroll;
    }
  }
  &-save {
    &Button {
      display: flex;
      justify-content: center;
      margin-top: 8px;
    }
  }
  @media (min-width: 768px) {
    &-content {
      flex-direction: row;
      &Form {
        & span > div {
          min-height: 362px;
          &:last-child {
            margin-bottom: 8px;
          }
        }
      }
    }
  }
}
