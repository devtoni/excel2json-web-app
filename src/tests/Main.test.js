import React from 'react'
import renderer from 'react-test-renderer'
import { Main } from '../components/Main/Main';

describe('Main', () => {
    it('starts with an empty array for the jsonFile', () => {
        const tree = renderer.create(<Main />).toJSON
        expect(tree).toMatchSnapshot()
    })
})