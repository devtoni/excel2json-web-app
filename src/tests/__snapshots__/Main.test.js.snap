// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main starts with an empty array for the jsonFile 1`] = `[Function]`;
